### ISSUE TITLE
Surface graphql function to plugins

### ISSUE THREAD

**Issue by cco3** [gridsome/gridsome#715]
It'd be nice if the api gave access to the graphql function.

### Basic example

```
api.afterBuild(async ({ config, graphql }) => {
  const data = await generatePodcast(graphql)
  const outFile = path.join(config.outDir, 'podcast.xml')
  await fs.writeFile(outFile, data)
})
```

### Motivation

It seems like several of the api functions could benefit from access to the actual site data, including the podcast generating example above.

**Comment by hjvedvik** [comment]
Yes, it would be useful to have access to the GraphQL schema after the build. I'll add this to the todo-list :)

**Comment by platform-kit** [comment]
@hjvedvik Is there another thread for discussing this idea, or shall we do it here?

I would love to write a plugin that exposes a Graphql API, ideally via a serverless script so it could be easily (zero-config) deployed on Netlify. Just not sure where to start.
